include/Server.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "Packet.hpp"
+
+#include <cstdint>
+#include <vector>
+#include <queue>
+
+/**
+ * Server receiver
+ *
+ * The main task of ServerBase -- receive packets and send ACKs
+ */
+struct ServerBase {
+    explicit ServerBase(std::uint32_t id);
+    ServerBase() = default;
+    virtual ~ServerBase() = default;
+
+    /**
+     * Get ID of server
+     * @return std::uint32_t -- server's ID
+     */
+    std::uint32_t GetID() const;
+
+    /**
+     * Get server's current time in nanoseconds
+     * @return std::uint64_t -- server's current time in nanoseconds
+     */
+    std::uint64_t GetCurrentTime() const;
+
+    /**
+     * Replace server's current time with a new value
+     * @param new_time -- new value of time in nanoseconds
+     */
+    void SetCurrentTime(std::uint64_t new_time);
```

Comment:
May I, please, ask not to use getters/setters at this stage. See the opinion on this here https://youtu.be/uHSLHvWFkto?t=4380 discussion and further discussion https://stackoverflow.com/questions/32625274/public-variables-bad-practice-vs-getters-and-setters-functions.